src/App.js

```diff
@@ -107,11 +107,22 @@ class App extends Component {
     if (this.hasWhitelistedScenes()) {
       this.requestScenesStatus();
     }
+    var intervalId = setInterval(this.timer, 5000);
+    // store intervalId in the state so it can be accessed later:
+    this.setState({intervalId: intervalId});
   }
 
   componentWillUnmount() {
     this.json.setEventHandler(() => {});
     this.mqtt.setEventHandler(() => {});
+    // use intervalId from the state to clear the interval
+    clearInterval(this.state.intervalId);
+  }
+
+  timer=()=>{
+    for (let i = 0; i < this.state.whitelist.length; i++) {
+      this.requestDeviceStatus(this.state.whitelist[i]);
+    }
```

Comment:
you can use [Array.forEach](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) here, eg `this.state.whitelist.forEach(device => this.requestDeviceStatus(device));`

Reply:
OK, will try this. But this is used several times in that way in the existing code. Maybe it should be changed there to (but not with this PullRequest)?

Reply:
It's just syntactic sugar, either way is fine. The latter is just more modern.

Reply:
done